$NetBSD: patch-CMakeLists.txt,v 1.4 2023/11/15 16:49:18 nia Exp $

CMake's detection of libsodium is broken for whatever reason,
and it finds uses libpthread. Use pkg-config to find it.

Do not force libstdc++ to be linked statically, creates problems
with PIE code.

--- CMakeLists.txt.orig	2023-11-15 16:40:30.182662267 +0000
+++ CMakeLists.txt
@@ -158,7 +158,8 @@ endif()
 if(NOT NONET)
   if(DEVILUTIONX_SYSTEM_LIBSODIUM)
     set(sodium_USE_STATIC_LIBS ${DEVILUTIONX_STATIC_LIBSODIUM})
-    find_package(sodium REQUIRED)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(sodium libsodium)
   else()
     add_subdirectory(3rdParty/libsodium)
   endif()
@@ -607,10 +608,6 @@ foreach(target devilution ${BIN_TARGET} 
   target_compile_definitions(${target} PRIVATE ${def_list})
 endforeach(target)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT HAIKU AND NOT VITA)
-  target_link_libraries(${BIN_TARGET} PUBLIC "$<$<NOT:$<CONFIG:Debug>>:-static-libgcc;-static-libstdc++>")
-endif()
-
 if(WIN32)
   target_link_libraries(${BIN_TARGET} PRIVATE wsock32 ws2_32 wininet)
 
